mesonbuild/wrap/wrap.py

```diff
@@ -817,17 +817,17 @@ def apply_diff_files(self) -> None:
             if not path.exists():
                 raise WrapException(f'Diff file "{path}" does not exist')
             relpath = os.path.relpath(str(path), self.dirname)
-            if GIT:
-                # Git is more likely to be available on Windows and more likely
-                # to apply correctly assuming patches are often generated by git.
-                # See https://github.com/mesonbuild/meson/issues/12092.
-                # The `--work-tree` is necessary in case we're inside a
+            if PATCH:
+                # Always pass a POSIX path to patch, because on Windows it's MSYS
+                # Ignore whitespace when applying patches to workaround
+                # line-ending differences
+                cmd = [PATCH, '-l', '-f', '-p1', '-i', str(Path(relpath).as_posix())]
+            elif GIT:
+                # If the `patch` command is not available, fall back to `git
+                # apply`. The `--work-tree` is necessary in case we're inside a
                 # Git repository: by default, Git will try to apply the patch to
                 # the repository root.
                 cmd = [GIT, '--work-tree', '.', 'apply', '-p1', relpath]
```

Comment:
I would add `--ignore-whitespace` here too for consistency then. Ideally also have a unit test that mix `\r\n`, I should have done one in the original PR tbh.

Reply:
Yeah, makes sense. I'll add it.